libmogwai-schedule-client/scheduler.c

```diff
@@ -1025,3 +1087,33 @@ mwsc_scheduler_schedule_entries_finish (MwscScheduler  *self,
 
   return g_task_propagate_pointer (G_TASK (result), error);
 }
+
+/**
+ * mwsc_scheduler_get_allow_downloads:
+ * @self: a #MwscScheduler
+ *
+ * Get the value of #MwscScheduler:allow-downloads.
+ *
+ * Returns: %TRUE if the user has indicated that at least one of the active
+ *    network connections should be used for large downloads, %FALSE otherwise
+ * Since: 0.1.0
+ */
+gboolean
+mwsc_scheduler_get_allow_downloads (MwscScheduler *self)
+{
+  g_return_val_if_fail (MWSC_IS_SCHEDULER (self), TRUE);
+
+  if (!check_invalidated (self, NULL))
+    return TRUE;
+
+  g_autoptr(GVariant) allow_downloads_variant = NULL;
+  allow_downloads_variant = g_dbus_proxy_get_cached_property (self->proxy, "DownloadsAllowed");
+
+  if (allow_downloads_variant == NULL)
+    {
+      g_debug ("%s: Could not get cached DownloadsAllowed property", G_STRFUNC);
```

Comment:
I was wondering whether we should return a `G_IO_NOT_FOUND` here instead, and let the caller decide what to do. But I guess returning the default is also fine...

Reply:
This branch should never really be taken — the proxy should always have a populated property cache. I should probably change that `g_debug()` into a `g_critical()`. I’ll change that.

Accordingly, since it’s a programmer error if the cache isn’t populated, a `GError` here would not be appropriate.